Add Property to ignore custom FBX Axis rotation

dellis1972 opened the pull request:
Fixes https://github.com/MonoGame/MonoGame/issues/8770
Context https://github.com/assimp/assimp/pull/5754

Assimp added a property to ignore custom FBX Axis rotation. But it was turned off by default. For MonoGame we want it on by default to maintain consistency.

dellis1972 committed:
Add Property to ignore custom FBX Axis rotation

Fixes https://github.com/MonoGame/MonoGame/issues/8770
Context https://github.com/assimp/assimp/pull/5754

Assimp added a property to ignore custom FBX Axis rotation.
But it was turned off by default. For MonoGame we want it on
by default to maintain consistency.

dellis1972 committed:
Fix test

dellis1972 committed:
Run the SpaceshipModel test

dellis1972 committed:
Reduce diff for Spaceship model test to 0.97

dellis1972 committed:
Reduce diff for Spaceship model test to 0.96

dellis1972 committed:
Ignore on Linux for now

dellis1972 committed:
Try a fixed timestep

dellis1972 committed:
try this instead

dellis1972 committed:
ff

TheKrumpet commented:
Had a quick test of this against my repro project and it appears to have resolved #8770 and is once again consistent with 3.8.2.x.

3.8.2.x:
<img width="802" height="512" alt="image" src="https://github.com/user-attachments/assets/e828154f-a37a-4648-95cd-c58a5c483833" />

This branch:
<img width="802" height="512" alt="image" src="https://github.com/user-attachments/assets/96ebf8ac-4f87-4f18-84f9-321631dab06b" />

tomspilman commented on Tools/MonoGame.Tools.Tests/FbxImporterTests.cs:
```diff
@@ -93,7 +93,7 @@ public void Dude()
 #if XNA
             Assert.AreEqual(11433, meshContent.Positions.Count);
 #else
-            Assert.AreEqual(13126, meshContent.Positions.Count);
```
How did this change @dellis1972 ?  What happened?

dellis1972 replied:
We bumped assimp. I haven't had the time to dig into why we now get more "Positions" back.

tomspilman replied:
I would suspect some weld tolerance or check for invalid triangles was change in AssImp.

Maybe we should double check this before merging?

dellis1972 replied:
Issue raised on assimp https://github.com/assimp/assimp/issues/6367

dellis1972 replied:
Added A comment

SimonDarksideJ commented:
Noted, there are additional "Import" issues being raised regarding Assimp model import now:
https://github.com/MonoGame/MonoGame/discussions/8985

tomspilman commented:
> Noted, there are additional "Import" issues being raised regarding Assimp model import now: #8985

Is it "additional"?  The reporter in #8985 said they were on 3.8.4.   This fix was merged last week... so not in the 3.8.4 release.

I think we should be sure the reporter tests against the latest preview pipeline with this fix in it.

dellis1972 committed:
Adjust expected position count in FbxImporterTests

Updated expected position count in FbxImporterTests due to Assimp changes.

dellis1972 committed:
Change similarity threshold to 97% in test

Adjusted similarity threshold for multi-frame test to account for floating point inaccuracies.

dellis1972 committed:
Update similarity threshold in SpaceshipModel test

Only run the spaceshipmodel test on windows for now.

rejurime commented:
Hi! Is this PR planned for inclusion in MG 3.8.5?

dellis1972 commented:
@rejurime 

I'm afraid we are still possibly blocked on https://github.com/assimp/assimp/issues/6367. More thumbs up on that issue might help

dellis1972 commented:
Looks like v6.0.4 might have a fix  https://github.com/assimp/assimp/pull/6278

We now are waiting on AssimpNetter to bump to v6.0.4 https://github.com/Saalvage/AssimpNetter/issues/26

dellis1972 committed:
Update AssimpNetter package version to 6.0.4.0

dellis1972 committed:
Fix expected position count in FbxImporterTests

dellis1972 commented:
@TheKrumpet  so we bumped to v6.0.4 of assimp which seems to have fixed the issue.

rejurime commented:
Great news! Happy to test it when it’s ready 💪

dellis1972 commented:
@TheKrumpet  any chance you'll have some time to test this works?

SimonDarksideJ commented:
@TheKrumpet any chance you can do a quick retest with your sample repo so we can sign off this fix and include it in `3.8.5`

vpenades commented:
Question: any chance this PR be merged for 3.8.5 final?

dellis1972 commented:
@vpenades it needs testing, I have repeatedly asked people who have this issue to test the fixes, but no one has reported if it works or not yet. So we are blocked until we get a thumbs up

vpenades commented:
I think the issue that prevents people from testing is, correct me if I am wrong, that you need to compile the whole monogame project so you have the appropiate DLLs to add to the project.

It's been a long time since I compiled monogame locally, but my understanding is that now it has a lot more native code than before, if that's the case I might not be able to compile it since I don't install C++ tooling anymore.

Something that could be interesting to add to monogame to address issues like this is to generate nuget packages for PRs... for example, it's what Avalonia does.

Anyway, if you make the nugets for this PR available somewhere I can give it a try

SimonDarksideJ commented:
> I think the issue that prevents people from testing is, correct me if I am wrong, that you need to compile the whole MonoGame project so you have the appropriate DLLs to add to the project.

Not strictly true, we have for some time now been shipping a "Binaries" build with every automation run, so if you check the action for this PR you will find an artefact called "[MonoGame.3.8.4.3787-develop](https://github.com/MonoGame/MonoGame/actions/runs/25013806430/artifacts/6669357956)"

https://github.com/MonoGame/MonoGame/actions/runs/25013806430?pr=8963

Granted, I suppose we should rename this to state it is the binaries release.

Alternatively, you can just clone and add MG as a project source, but this can be more complicated.

dellis1972 commented:
@vpenades we do publish NuGet's to the Github feed. Its documented how to use them https://docs.monogame.net/articles/getting_started/using_development_nuget_packages.html. 

Its just a pain to setup the access tokens and everything, I don't think anyone bothers. 
Does Avalonia publish PR nugets to actual Nuget? or just the GitHub feed?

uygary commented:
I think adding an automation to post a message (here in the PR) containing the links to the packages would be enough for discoverability. Since it's a bit buried right now, it's not all too obvious, so adding that might be good.
I fear publishing these to the official nuget source would cause nothing but confusion for most folks.

Anyone who wants it more automated could use the GitHub feed.

dellis1972 committed:
Merge branch 'develop' into dellis1972/assimp6

SimonDarksideJ committed:
Merge branch 'develop' into dellis1972/assimp6

SimonDarksideJ reviewed (approved):
Good to go @dellis1972 , should we get this merged?
